### ISSUE TITLE
Wrong RCS1206 when argument is an Expression<Func<T...>>

### ISSUE THREAD

**Issue by WebFreak001** [JosefPihrt/Roslynator#691]
**Product and Version Used**: 

VSCode 1.47.0 with Roslynator 3.0.0
.NET Core 3.1.301 with language version 8.0 (nullables enabled)

**Steps to Reproduce**:
Given the following code:
```cs
using System;
using System.Linq.Expressions;

public class Program
{
	public static void Main()
	{
		Foo(a => a.HasValue ? (DateTime?)a.Value.ToUniversalTime() : null);
	}

	public static void Foo(Expression<Func<DateTime?, DateTime?>> f)
	{
		f.Reduce();
	}
}
```

**Actual Behavior**:

RCS1206 is triggered on line 8 (`Use conditional access instead of conditional expression`), rewriting the Main function to:

```csharp
	public static void Main()
	{
		Foo(a => (a?.ToUniversalTime()));
	}
```

Causing a compiler error CS8072 `An expression tree lambda may not contain a null propagating operator.`

**Expected Behavior**:

No refactoring should be offered as null propagation inside an Expression<...> argument is unsupported by the language.
